view/src/components/farms.js

```diff
@@ -0,0 +1,819 @@
+  },
+  closeButton: {
+    position: "absolute",
+    right: theme.spacing(1),
+    top: theme.spacing(1),
+    color: theme.palette.grey[500],
+  },
+  search: {
+    position: "relative",
+    borderRadius: theme.shape.borderRadius,
+    marginLeft: 0,
+    width: "100%",
+  },
+  searchIcon: {
+    padding: theme.spacing(0, 2),
+    height: "100%",
+    position: "absolute",
+    pointerEvents: "none",
+    display: "flex",
+    alignItems: "center",
+    justifyContent: "center",
+  },
+  inputRoot: {
+    color: "inherit",
+  },
+  inputInput: {
+    padding: theme.spacing(1, 1, 1, 0),
+    paddingLeft: `calc(1em + ${theme.spacing(4)}px)`,
+    width: "100%",
+  },
+  chip: {
+    margin: "4px",
+  },
+});
+
+const Transition = React.forwardRef(function Transition(props, ref) {
+  return <Slide direction="up" ref={ref} {...props} />;
+});
+
+/**
+ * This function sets up the mask used for the phone input of (***) ***-****
+ * where * represents a digit
+ */
+function TextMaskCustom(props) {
+  const { inputRef, ...other } = props;
+
+  return (
+    <MaskedInput
+      {...other}
+      ref={(ref) => {
+        inputRef(ref ? ref.inputElement : null);
+      }}
+      mask={[
+        "(",
+        /[1-9]/,
+        /\d/,
+        /\d/,
+        ")",
+        " ",
+        /\d/,
+        /\d/,
+        /\d/,
+        "-",
+        /\d/,
+        /\d/,
+        /\d/,
+        /\d/,
+      ]}
+      placeholderChar={"\u2000"}
+      showMask
+    />
+  );
+}
+
+TextMaskCustom.propTypes = {
+  inputRef: PropTypes.func.isRequired,
+};
+
+/**
+ * This class represents a Farm component, which is a sub-page of the
+ * home page where farm objects are visualized, created, updated, edited,
+ * and deleted.
+ */
+class Farms extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      // farm states
+      farmName: "",
+      farmId: "",
+      contactName: "",
+      contactEmail: "",
+      contactPhone: "(1  )    -    ",
+      farmTags: [],
+      forklift: false,
+      loadingDock: false,
+      location: "",
+      locationId: "",
+      transportation: false,
+      // page states
+      errors: [],
+      open: false,
+      uiLoading: true,
+      buttonType: "",
+      viewOpen: false,
+    };
+
+    this.onTagsChange = this.onTagsChange.bind(this);
+    this.handleDelete = this.handleDelete.bind(this);
+    this.handleEditClick = this.handleEditClick.bind(this);
+    this.handleViewOpen = this.handleViewOpen.bind(this);
+  }
+
+  /**
+   * Given an event, this function updates a state (the target of the event)
+   * with a new value
+   * @param event The event that is attempting to update a state
+   */
+  handleChange = (event) => {
+    this.setState({
+      [event.target.name]: event.target.value,
+    });
+  };
+
+  /** Used to uncheck and check the form checkboxes */
+  handleChecked = (name) => (event) => {
+    this.setState({ ...this.state, [name]: event.target.checked });
```

Comment:
Déjà vu again with this comment, but why do you need `...this.state` if setState automatically merges?

Reply:
I forgot to change it.... tomorrow I will go through every file and make sure its changed